server/routes.go

```diff
@@ -1640,3 +1642,23 @@ func handleScheduleError(c *gin.Context, name string, err error) {
 		c.JSON(http.StatusInternalServerError, gin.H{"error": err.Error()})
 	}
 }
+
+var thinkTagRegexp = regexp.MustCompile(`<think>(?s).*?</think>(\n)*`)
+
+func filterThinkTags(msgs []api.Message, model *Model) []api.Message {
+	if model.Config.ModelFamily == "qwen3" || strings.HasPrefix(model.ShortName, "deepseek-r1:") {
```

Comment:
This works – to get the "model" component of the name you could also use `name.Model` similar to here: https://github.com/ollama/ollama/blob/7b68e254c29bea48dec9a455a5a46b4ac6c823c2/server/routes.go#L1483

Reply:
updated, just pushed a new commit, please see if that's what you were hoping for